File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/Escapeshellarg.java
package com.onevgo.functions;

public class Escapeshellarg {
    public static String escapeshellarg(String arg) {
        return "'" + arg.replace("'", "'\\\''") + "'";
    }

    public static void main(String[] args) {
        System.out.println("ls " + escapeshellarg("./"));
        System.out.println("ls " + escapeshellarg("','"));
    }
}

File: 01_Language/01_Functions/js/escapeshellarg.js
function escapeshellarg(arg) {
    return arg.split("'").map(function (value, index, array) {
        return "'" + value + "'"
    }).join("\\'")
}

console.log('ls ' + escapeshellarg("./"))
console.log('ls ' + escapeshellarg("','"))
